master/internal/db/postgres_trial_metrics.go

```diff
@@ -24,6 +26,71 @@ const (
 	GenericMetric MetricPartitionType = "GENERIC"
 )
 
+type metricsBody struct {
```

Comment:
open to suggestions for a more go-like interface for serializing and deserializing these considering the structure changes based on metric type.